services/tone_analyzer/v3.js

```diff
@@ -0,0 +1,163 @@
+  // service credentials. This separation from sUsername and username to allow 
+  // the end user to modify the node credentials when the service is not bound.
+  // Otherwise, once set username would never get reset, resulting in a frustrated
+  // user who, when he errenously enters bad credentials, can't figure out why
+  // the edited ones are not being taken.
+
+  // Not ever used, and codeacy complains about it.
+
+  var username, password, sUsername, sPassword;
+
+  var service = cfenv.getAppEnv().getServiceCreds(/tone analyzer/i)
+
+  if (service) {
+    sUsername = service.username;
+    sPassword = service.password;
+  }
+
+  // Node RED Admin - fetch and set vcap services
+  RED.httpAdmin.get('/watson-tone-analyzer/vcap', function (req, res) {
+    res.json(service ? {bound_service: true} : null);
+  });
+
+
+  // Function that checks the configuration to make sure that credentials,
+  // payload and options have been proviced in the correct format.
+  var checkConfiguration = function(msg, config, node, cb) {
+    if (cb) {
+      var message = '';      
+      var isBuffer = false;
+
+      // Credentials are needed for each of the modes.
+      username = sUsername || node.credentials.username;
+      password = sPassword || node.credentials.password;      
+
+      if (!username || !password) {
+        message = 'Missing Tone Analyzer service credentials';
+        cb(message, null);
+        return;
+      }
+
+      if (!msg.payload) {
+        message = 'Missing property: msg.payload';
+        cb(message, null);
+        return;
+      } else {
+        var hasJSONmethod = (typeof msg.payload.toJSON === 'function') ;
+
+        if (hasJSONmethod === true) {
+          if (msg.payload.toJSON().type === 'Buffer') {
+            isBuffer = true;
+          }      
+        }      
+      }
+
+      // Payload (text to be analysed) must be a string (content is either raw string or Buffer)
+      if (typeof msg.payload !== 'string' &&  isBuffer !== true) {
+        message = 'The payload must be either a string or a Buffer';
+        cb(message, null);
+        return;
+      }
+
+      var tones = msg.tones || config.tones;
+      var sentences = msg.sentences || config.sentences;
+      var contentType = msg.contentType || config.contentType
+
+      var taSettings = {
+        'username' : username, 
+        'password' : password,
+        'tones' : tones,
+        'sentences' : sentences,
+        'contentType' : contentType
+      };
+
+      cb(null, taSettings);
+    }
+  };
+
+
+  // function when the node recieves input inside a flow. 
+  // Configuration is first checked before the service is invoked.
+  var processOnInput = function(msg, config, node) {
+    checkConfiguration (msg, config, node, function(err, settings){
+      if (err) {
+        node.status({fill:'red', shape:'dot', text:err}); 
+        node.error(err, msg);
+        return;
+      }
+      else {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Closing curly brace does not appear on the same line as the subsequent block.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2652984208/issues/source?bid=3225115&fileBranchId=3325196#l110)